### PR TITLE
Add more diagnostics for TestKubernetesTLSRotation test

### PR DESCRIPTION
**Pull request by sivanov-nuodb**
*Changes*
- Disable CoreDNS entries caching (https://kubernetes.io/docs/tasks/administer-cluster/dns-custom-nameservers/)
- Print pods if `TestKubernetesTLSRotation` test fails using diagnostics teardown
- General ability to inject arbitrarily custom values file `valuesInject.yaml` for all tests. This is useful in case there is need to overwrite some of the inline values like admin config files or affinity rules.

### CODE REVIEW

File: test/minikube/minikube_tls_rotation_test.go [hunk 1]
```diff
@@ -116,10 +116,16 @@ func TestKubernetesTLSRotation(t *testing.T) {
 	testlib.CreateSecret(t, namespaceName, testlib.CA_CERT_FILE, testlib.CA_CERT_SECRET, newTLSKeysLocation)
 	testlib.CreateSecretWithPassword(t, namespaceName, testlib.KEYSTORE_FILE, testlib.KEYSTORE_SECRET, testlib.SECRET_PASSWORD, newTLSKeysLocation)
 
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+	testlib.AddDiagnosticTeardown(testlib.TEARDOWN_ADMIN, t, func() {
+		output, _ := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "get", "pods", "-o", "wide")
+		t.Logf("All pods in namespace=%s:", namespaceName)
+		t.Logf(output)
```
[the remark this answers is not in the document]
Comment (sivanov-nuodb) [reply test/minikube/minikube_tls_rotation_test.go:123]: The `t.Logf()` worked for me but I agree that `t.Log` would be more appropriate here.

File: test/minikube/minikube_tls_rotation_test.go [hunk 2]
```diff
@@ -116,10 +116,16 @@ func TestKubernetesTLSRotation(t *testing.T) {
 	testlib.CreateSecret(t, namespaceName, testlib.CA_CERT_FILE, testlib.CA_CERT_SECRET, newTLSKeysLocation)
 	testlib.CreateSecretWithPassword(t, namespaceName, testlib.KEYSTORE_FILE, testlib.KEYSTORE_SECRET, testlib.SECRET_PASSWORD, newTLSKeysLocation)
 
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+	testlib.AddDiagnosticTeardown(testlib.TEARDOWN_ADMIN, t, func() {
+		output, _ := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "get", "pods", "-o", "wide")
+		t.Logf("All pods in namespace=%s:", namespaceName)
```
[the remark this answers is not in the document]
Comment (sivanov-nuodb) [reply test/minikube/minikube_tls_rotation_test.go:122]: Sure. I will shorten this to just `k8s.RunKubectl` which should print the output on the console.

File: scripts/ci/install_deps.sh [hunk 3]
```diff
@@ -21,6 +21,12 @@ if [[ -n "$REQUIRES_MINIKUBE" ]]; then
   sudo chown -R travis: /home/travis/.minikube/
   kubectl cluster-info
 
+  # In some tests (specifically TestKubernetesTLSRotation), we observe incorrect DNS resolution 
+  # after pods have been re-created which causes problems with inter pod communicataion.
+  # Set CoreDNS TTL to 0 so that DNS entries are not cached. 
+  kubectl get cm coredns -n kube-system -o yaml | sed -e 's/ttl [0-9]*$/ttl 0/' | kubectl apply -n kube-system -f -
```
Comment (adriansuarez) [inline scripts/ci/install_deps.sh:27]: Have looked at `kubectl patch`? This would be more structured and less brittle.

Comment (sivanov-nuodb) [reply scripts/ci/install_deps.sh:27]: @adriansuarez I normally use `kubectl patch` when there is a need to change something specific in K8s resource spec. AFAIK with configmaps, you can patch the `data: {config: "MULTILINE TEXT"}` and not specifically the `ttl` property in the text. I didn't want to overwrite the full CoreDNS config as it might change in future.

**Review by acabrele — APPROVED** [review]
